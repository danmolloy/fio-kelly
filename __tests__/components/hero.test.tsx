import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import HeroSection from "../../components/hero";

describe("<HeroSection />", () => {
  beforeEach(() => {
    render(<HeroSection />);
  });
  it("hero-section is in the document", () => {
    const heroSection = screen.getByTestId("hero-section");
    expect(heroSection).toBeInTheDocument();
  });
  it("'Fiona Kelly' title is in the document", () => {
    const title = screen.getByText("Fiona Kelly");
    expect(title).toBeInTheDocument();
  });
  it("portrait image is in the document", () => {
    const portraitImg = screen.getByTitle("Fiona Kelly");
    expect(portraitImg).toBeInTheDocument();
    expect(portraitImg).toHaveAttribute("alt", "Fiona Kelly");
  });
  it("matches snapshot", () => {
    const heroSection = screen.getByTestId("hero-section");
    expect(heroSection).toMatchSnapshot();
  });
});
