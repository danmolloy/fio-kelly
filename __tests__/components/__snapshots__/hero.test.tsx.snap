// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroSection /> matches snapshot 1`] = `
<div
  class=" w-screen h-screen flex flex-col justify-center overflow-hidden items-center"
  data-testid="hero-section"
>
  <div
    class="hidden md:flex z-10  flex-col absolute top-36  md:right-24"
  >
    <h1
      class=" text-zinc-200 text-5xl  font-display"
    >
      Fiona Kelly
    </h1>
  </div>
  <div
    class="overflow-hidden w-[925px] md:w-screen -ml-28 sm:-ml-0 -mt-0 lg:-mt-56 h-screen md:h-[1477px] mx-auto outline"
  >
    <img
      alt="Portrait of Fiona lounging on a couch with her flute."
      data-nimg="1"
      decoding="async"
      height="1477"
      loading="lazy"
      sizes=""
      src="/_next/image?url=%2Fimages%2Ffiona-chaise.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Ffiona-chaise.jpg&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Ffiona-chaise.jpg&w=3840&q=75 2x"
      style="color: transparent;"
      title="Portrait of Fiona lounging on a couch with her flute."
      width="1850"
    />
  </div>
</div>
`;
